### PR TITLE
fix(series): fix series data type judgment bug. close #14293

### LINKED ISSUE #14293: 配置项series.data数组里如果同时存在null和对象,将会报错Cannot read property 'selected' of null
### Version
5.0.2

### Steps to reproduce
非连续的折线图/柱状图,在3.x和4.x版本时,
series.data可以用类似[null,{value:3},{value:4},null]这样的数组呈现,而在5.x版本时候必须使用[{value:null},{value:3},{value:4},{value:null}]
否则将会报错Cannot read property 'selected' of null

### What is expected?
正常呈现非连续的折线图

### What is actually happening?
报错Cannot read property 'selected' of null

### PR DESCRIPTION
**Pull request by leosxie**
## Brief Information
fix series data type judgment bug

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

fix series data type judgment bug, typeof null === "object"

### Fixed issues
#14293 

## Details
when set 

### Before: What was the problem?
image
when rawItem is null, get prop selected error

### After: How is it fixed in this PR?
image

### Are there any API changes?

- [ ] The API has been changed.

### Related test cases or examples to use the new APIs

test/line-non-continuous.html

### Merging options

- [ ] Please squash the commits into a single one when merge.

### Other information

### CODE REVIEW

File: src/model/Series.ts [hunk 1]
```diff
@@ -575,8 +575,8 @@ class SeriesModel<Opt extends SeriesOption = SeriesOption> extends ComponentMode
         if (data.hasItemOption) {
             data.each(function (idx) {
                 const rawItem = data.getRawDataItem(idx);
-                if (typeof rawItem === 'object'
-                    && (rawItem as OptionDataItemObject<unknown>).selected
+                if (zrUtil.isObject(rawItem)
+                    && (rawItem as OptionDataItemObject<unknown>)?.selected
```
Comment (pissang) [inline src/model/Series.ts:579]: Optional chaining will be transpiled to
```js
rawItem === null || rawItem === void 0 ? void 0 : rawItem.selected;
```
and brings extra codes in the final distribution.

So we limited the use of optional chaining in echarts to keep the distribution as small as possible.

Instead 
```ts
(rawItem && (rawItem as OptionDataItemObject<unknown>).selected)
```
is enough here.

**Review by pissang — CHANGES_REQUESTED** [review]

File: src/model/Series.ts (hunk 1 above)
Comment (leosxie) [reply src/model/Series.ts:579]: get it

Comment (leosxie) [reply src/model/Series.ts]: use zrUtil.isObject(rawItem) replace in here may be more safe?

File: src/model/Series.ts [hunk 2]
```diff
@@ -575,8 +575,7 @@ class SeriesModel<Opt extends SeriesOption = SeriesOption> extends ComponentMode
         if (data.hasItemOption) {
             data.each(function (idx) {
                 const rawItem = data.getRawDataItem(idx);
-                if (typeof rawItem === 'object'
-                    && (rawItem as OptionDataItemObject<unknown>).selected
+                if (rawItem && (rawItem as OptionDataItemObject<unknown>).selected
                 ) {
```
Comment (pissang) [inline src/model/Series.ts:579]: No need to newline here for a better code style.
```ts
if (rawItem && (rawItem as OptionDataItemObject<unknown>).selected) {
```

**Review by pissang — CHANGES_REQUESTED** [review]

File: .history/src/model/Series_20210308122052.ts [hunk 3]
```diff
@@ -0,0 +1,693 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
```
Comment (pissang) [inline .history/src/model/Series_20210308122052.ts:2]: This file should not be committed

**Review by pissang — CHANGES_REQUESTED** [review]

**Review by pissang — APPROVED** [review]

**Pull request merged**